src/main/java/com/google/devtools/build/lib/runtime/QueryRuntimeHelper.java

```diff
@@ -91,6 +108,34 @@ public void afterQueryOutputIsWritten() {}
       @Override
       public void close() {}
     }
+
+    private static class FileQueryRuntimeHelper implements QueryRuntimeHelper {
+      private final Path path;
+      private final OutputStream out;
+
+      public FileQueryRuntimeHelper(Path path) throws IOException {
+        out = path.getOutputStream();
+        this.path = path;
+      }
+
+      @Override
+      public OutputStream getOutputStreamForQueryOutput() {
+        return out;
+      }
+
+      @Override
+      public void afterQueryOutputIsWritten() {}
```

Comment:
Should we `close` the file stream here instead of in `close`? I'm not sure when this is called or what it's supposed to do.

Reply:
An internal implementation calls `close()` in here and has more followup steps after.

But for Bazel, `close()` should be sufficient because the callsite uses `try-with-resources` for this object.